lib/configuration/variables/sources/merge.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const ServerlessError = require('../../../serverless-error');
+
+module.exports = {
+  resolve: ({ params }) => {
+    if (!Array.isArray(params) || params.length < 2) {
+      throw new ServerlessError('${merge(...)} needs a sequence of values to merge');
+    }
+
+    if (params.every((item) => Array.isArray(item))) {
+      return {
+        value: params.reduce((acc, item) => {
+          acc.push(...item);
+          return acc;
+        }, []),
```

Comment:
Let's maybe use normal iteration instead (using `reduce` feels controversial, as it's not a real use case for reduce)

I'd put it as:

```javascript
const result = [];
for (const param of params) result.push(...param)
return { value: result };
```